apps/api/src/app/inbox/dtos/snooze-notification-request.dto.ts

```diff
@@ -0,0 +1,40 @@
+import { IsDate, ValidationArguments, registerDecorator, ValidationOptions } from 'class-validator';
+import { Type } from 'class-transformer';
+
+function IsFutureDate(
+  options?: {
+    minBufferMs?: number;
+  },
+  validationOptions?: ValidationOptions
+) {
+  const minBufferMs = options?.minBufferMs ?? 1000 * 60; // default 1 minute
+
+  return function (object: Object, propertyName: string) {
```

Comment:
Lovely! You can use https://date-fns.org/v4.1.0/docs/isFuture and some a few lines of code.